### PR TITLE
Add CountDown Timer in challenge phase

### PR DESCRIPTION
**Pull request by RishabhJain2018**
This PR adds a countdown timer for the challenge phases. The timer will automatically start when the challenge will start.

@deshraj @isht3 @live-wire Please review the PR.

### CODE REVIEW

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -343,8 +343,29 @@
...
+                        var challengePhaseEndDate = Date.parse(vm.phases.results[i].end_date);
+                        var dateNow = Date.now();
+                        var timeDiff = parseInt((challengePhaseEndDate - dateNow)/1000);
+
+                        if (timeDiff > 0 && dateNow < challengePhaseEndDate) {
+                            vm.countDownTime = timeDiff;
+                            vm.phases.results[i].showCountdownTimer=true;
+                        }
+
+                        // The below code can used to make a coundown timer that will start one day before a challenge deadline
```
Comment (deshraj) [inline frontend/src/js/controllers/challengeCtrl.js]: Please remove the commented code if not being used.

File: frontend/src/views/web/challenge/phases.html [hunk 2]
```diff
@@ -3,9 +3,13 @@
         <div class="row">
             <div class="col s11 m11">
                 <div class="row">
-                    <div class="col s12">
+                    <div class="col s4">
```
Comment (deshraj) [inline frontend/src/views/web/challenge/phases.html]: I am not sure how this will look like? can you please add a GIF/Video showing the timer?

**Comment by deshraj** [comment]
> The timer will start automatically one day before the challenge phase ends.

Why not show the timer when the challenge starts?

**Comment by RishabhJain2018** [comment]
Yes! I have updated the PR to show the timer when the challenge starts. I forgot to update the description.

File: frontend/src/views/web/challenge/phases.html (hunk 2 above)
Comment (RishabhJain2018) [reply frontend/src/views/web/challenge/phases.html]: I have intentionally left this code if in future we might need it. Moreover, the video is at the channel on the slack.

Comment (deshraj) [reply frontend/src/views/web/challenge/phases.html]: > I have intentionally left this code if in future we might need it. 

Let's not keep the commented code here. We can always implement it. It's quite simple and straightforward. 

> Moreover, the video is at the channel on the slack.

The PRs are public and we should make sure that all the PR related details are there in the PR description. :)

Comment (RishabhJain2018) [reply frontend/src/views/web/challenge/phases.html]: > Let's not keep the commented code here. We can always implement it. It's quite simple and straightforward.
Sure! I'll remove it and update the PR.

> The PRs are public and we should make sure that all the PR related details are there in the PR description. :)
I will add this too!!

File: frontend/src/js/controllers/challengeCtrl.js (hunk 1 above)
Comment (deshraj) [reply frontend/src/js/controllers/challengeCtrl.js]: This is not resolved.

File: frontend/src/views/web/challenge/phases.html (hunk 2 above)
Comment (deshraj) [reply frontend/src/views/web/challenge/phases.html]: This is also not done. Please let me know when are done with all the above cahnges.

**Comment by deshraj** [comment]
@RishabhJain2018 can you please add a screenshot here?

**Comment by RishabhJain2018** [comment]
Closing this PR!

**Pull request closed without merging**